sim/scenarios/droplist/droplist-error-model.cc

```diff
@@ -0,0 +1,35 @@
+#include <map>
+#include "droplist-error-model.h"
+
+using namespace std;
+
+NS_OBJECT_ENSURE_REGISTERED(DroplistErrorModel);
+ 
+TypeId DroplistErrorModel::GetTypeId(void) {
+  static TypeId tid = TypeId("DroplistErrorModel")
+    .SetParent<ErrorModel>()
+    .AddConstructor<DroplistErrorModel>()
+    ;
+  return tid;
+}
+ 
+DroplistErrorModel::DroplistErrorModel() { }
+
+void DroplistErrorModel::DoReset(void) { }
+ 
+bool DroplistErrorModel::DoCorrupt(Ptr<Packet> p) {
+  static int packet_num = 0;
+  return drops.empty() ? 0 : drops.find(++packet_num) != drops.end();
+}
+
+void DroplistErrorModel::SetDrops(string drops_in) {
```

Comment:
I'd prefer to do the parsing in the main fail, and pass in a single packet number here, e.g.
`DroplistErrorModel::SetDrop(int packet_number)`.

Reply:
Good suggestion -- I've moved it into droplist.cc.